package/MDAnalysis/coordinates/base.py

```diff
@@ -1363,14 +1364,21 @@ def add_auxiliary(self, auxname, auxdata, **kwargs):
         Auxiliary data is assumed to be time-ordered, with no duplicates. See 
         the :ref:`Auxiliary API`.
         """
+        # making auxname and auxdata keyworded so we can more easily reload
+        # an auxiliary; so we now have to check we have both
+        if not auxname or not auxdata:
+            raise TypeError("Must provide an auxname and auxdata")
+ 
         if auxname in self.aux_list:
             raise ValueError("Auxiliary data with name {name} already "
                              "exists".format(name=auxname))
         if isinstance(auxdata, AuxReader):
             aux = auxdata
+            aux.auxname = auxname
         else:
-            auxreader = get_auxreader_for(auxdata)
-            aux = auxreader(auxdata, name=auxname, **kwargs)
+            auxreader = get_auxreader_for(auxdata, 
+                                          format=kwargs.pop('format', None))
```

Comment:
Do not forget to mention in the documentation of `AuxReader` that an auxiliary reader cannot expect a `format` keyword. If it does, it might create a conflict here.